nodejs/eks/cert-thumprint.ts

```diff
@@ -40,10 +40,12 @@ export function getIssuerCAThumbprint(issuerUrl: pulumi.Output<string>): pulumi.
 // Find the root CA cert in a chain of certs.
 function findCACertificate(certificate: tls.DetailedPeerCertificate): tls.DetailedPeerCertificate {
     let cert = certificate;
+    let issuerCert = cert?.issuerCertificate;
     for (; cert?.issuerCertificate?.fingerprint !== cert?.fingerprint;) {
+        issuerCert = cert;
```

Comment:
I'm not convinced this code is right. It looks like the goal is for `cert.issuerCertificate` to match `cert.fingerprint`?

Reply:
`cert` and `issuerCert` are placeholders that get overwritten as the chain of certs is traversed. 

The idea is to continuously view the fingerprint for the current cert's issuerCert. once they are equal, that means we have reached the root CA, and we return what was stored in `issuerCert`.

It is the non-recursive implementation used in [CDK](https://github.com/aws/aws-cdk/pull/6062/files#diff-0c88b2ccea9d77dbb7ac2285fd5f63afR114-R122)

Reply:
How about the following instead?

```ts
let cert = certificate;
while (cert.issuerCert && cert.issuerCert.fingerprint !== cert?.fingerprint) {
    cert = cert.issuerCert;
}
return cert;
```

[Playground Link](https://www.typescriptlang.org/v2/en/play?#code/C4TwDgpgBAwhBOwoF4oG8CwAoKuoDMBLAOwHMEx4TgB+ALigGdgqzs8pDHGBXBORPVgJg2AL7YAxgHtizAtOkMBSVJhx4iZCq2AMARAEN9AGnZ4uvfiIbqOHLeXiVqBgEanz9ztz7wVtl7emiROLsR6UPqSnhreEnFQCQnYADYQSJI2wogoCtIA3NgA7gAWhOlQABRZiAB0ln4qUABkLVC1wA2+1vWOOtRQAITIqJ00df3OugCU6EGdeZ3dVv4iRVgJMnLS6XWp0qRVAOQASopIU+GRx1AA1B0ik6EDETNAA)

Reply:
Turns out we have to return the intermediate root CA cert, not the trusted root CA to configure the OIDC provider.

The intermediate root CA cert is the _second to last_ cert in the full chain.

See the updated [changes](https://github.com/pulumi/pulumi-eks/pull/343/files) for more info and links.